src/wasm/wasm-binary.cpp

```diff
@@ -5169,7 +5275,10 @@ void WasmBinaryBuilder::visitRefFunc(RefFunc* curr) {
     throwError("ref.func: invalid call index");
   }
   functionRefs[index].push_back(curr); // we don't know function names yet
-  curr->finalize();
+  // To support typed function refs, we give the reference not just a general
+  // funcref, but a specific subtype with the actual signature.
+  curr->finalize(
+    Type(HeapType(getFunctionSignatureByIndex(index)), /* nullable = */ true));
```

Comment:
Shouldn't `ref.func` have a non-nullable type? I believe this should be the case even if we disallow non-nullable types everywhere else.

Reply:
As in the previous comment, I thought we agreed we would only support nullable types for now as a first step? Allowing them on `ref.func` would mean allowing them in enough places as to be as hard to support as to support them everywhere. (That is, a single `ref.func` introduces a non-nullable type, which then cannot be spilled to a local, which then...)

Reply:
I believe we resolved this offline. We agreed that it was simpler to pretend that even producers of non-nullable types produce nullable types for now, rather than apply that relaxation only to consumers of non-nullable types, as I had been envisioning.